code/datums/martial/sleeping_carp.dm

```diff
@@ -232,7 +234,7 @@
 		to_chat(user, "<span class='warning'>It would be dishonorable to attack a foe while they cannot retaliate.</span>")
 		return
 	if(user.a_intent == INTENT_DISARM)
-		if(!wielded)
+		if(!SEND_SIGNAL(src, COMSIG_IS_TWOHANDED_WIELDED))
```

Comment:
Two things:

All signals must return flags, do not treat the return like a simple bool. Look at how other signal returns are treated.

This is what we talked about where a GetComponent call is replaced by what's functionaly a GetComponent signal instead. It has the same issues. Use the wield and unwielded signals to keep track of the state.

Reply:
https://github.com/tgstation/tgstation/blob/41c850d6b3e8a65e2422d2642111bd9f85fcd3ab/code/game/objects/items.dm#L597

https://github.com/tgstation/tgstation/blob/368db973eb898f74f430a74f9895f6940ec7d104/code/__DEFINES/dcs/signals.dm#L367-L380

most of these are checked in this manner.

https://github.com/tgstation/tgstation/blob/ed9aca6f626cf2e15ccb353978ccb6d8d0b075a0/code/modules/jobs/access.dm#L14

Reply:
@ShizCalev okay but just because shitty code exists does not mean it should be baseline. Else we could say mecha code is the standard.

Reply:
Storage component is shit and those signals are all bad

Reply:
I will also say nanites do the exact same thing so that's two things that are shitty

Reply:
nanites either return flags or they modify an arg list

Reply:
How is this a flag? This is returning a bool check.

https://github.com/tgstation/tgstation/blob/645b339a8ca76801232c464ce48b48d13394976a/code/datums/components/nanites.dm#L221-L222

Reply:
This signal is gone, resolving